tests/testsuite/cargo_config/mod.rs

```diff
@@ -81,168 +82,149 @@ fn get_toml() {
-        .with_stderr("")
+        .with_stdout_data(str![[r#"
+profile.dev.opt-level = 3
+
+"#]])
+        .with_stderr_data(str![[r#""#]])
         .run();
 
     // A key that isn't set.
     cargo_process("config get build.rustc -Zunstable-options")
         .cwd(&sub_folder.parent().unwrap())
         .masquerade_as_nightly_cargo(&["cargo-config"])
         .with_status(101)
-        .with_stdout("")
-        .with_stderr("error: config value `build.rustc` is not set")
+        .with_stdout_data(str![[r#""#]])
+        .with_stderr_data(str![[r#"
+[ERROR] config value `build.rustc` is not set
+
+"#]])
         .run();
 
     // A key that is not part of Cargo's config schema.
     cargo_process("config get not.set -Zunstable-options")
         .cwd(&sub_folder.parent().unwrap())
         .masquerade_as_nightly_cargo(&["cargo-config"])
         .with_status(101)
-        .with_stdout("")
-        .with_stderr("error: config value `not.set` is not set")
+        .with_stdout_data(str![[r#""#]])
+        .with_stderr_data(str![[r#"
+[ERROR] config value `not.set` is not set
+
+"#]])
         .run();
 }
 
 #[cargo_test]
 fn get_json() {
-    // Notes:
-    // - This does not show env vars at all. :(
-    let all_json = r#"
-            {
-              "alias": {
-                "foo": "abc --xyz",
-                "sub-example": [
-                  "sub",
-                  "example"
-                ]
-              },
-              "build": {
-                "jobs": 99,
-                "rustflags": [
-                  "--flag-global",
-                  "--flag-directory"
-                ]
-              },
-              "extra-table": {
-                "somekey": "somevalue"
-              },
-              "profile": {
-                "dev": {
-                  "opt-level": 3,
-                  "package": {
-                    "foo": {
-                      "opt-level": 1
-                    }
-                  }
-                }
-              },
-              "target": {
-                "cfg(target_os = \"linux\")": {
-                  "runner": "runme"
-                }
-              }
-            }
-            "#;
     let sub_folder = common_setup();
     cargo_process("config get --format=json -Zunstable-options")
         .cwd(&sub_folder.parent().unwrap())
         .masquerade_as_nightly_cargo(&["cargo-config"])
         .env("CARGO_ALIAS_BAR", "cat dog")
         .env("CARGO_BUILD_JOBS", "100")
-        .with_json(all_json)
-        .with_stderr(
-            "\
-note: The following environment variables may affect the loaded values.
+        .with_stdout_data(r#"
+{"alias":{"foo":"abc --xyz","sub-example":["sub","example"]},"build":{"jobs":99,"rustflags":["--flag-global","--flag-directory"]},"extra-table":{"somekey":"somevalue"},"profile":{"dev":{"opt-level":3,"package":{"foo":{"opt-level":1}}}},"target":{"cfg(target_os = \"linux\")":{"runner":"runme"}}}
+
+"#.json_lines())
```

Comment:
`.json()` is sufficient. JSON lines are mainly for `--message-format=json` like output. Also, within a single JSON object we are supposed to be able to pretty print it. Pretty-printed JSON objects are easier to review.


```suggestion
{
  "alias": {
    "foo": "abc --xyz",
    "sub-example": [
      "sub",
      "example"
    ]
  },
  "build": {
    "jobs": 99,
    "rustflags": [
      "--flag-global",
      "--flag-directory"
    ]
  },
  "extra-table": {
    "somekey": "somevalue"
  },
  "profile": {
    "dev": {
      "opt-level": 3,
      "package": {
        "foo": {
          "opt-level": 1
        }
      }
    }
  },
  "target": {
    "cfg(target_os = \"linux\")": {
      "runner": "runme"
    }
  }
}

"#.json())
```

Reply:
Thanks for the detailed explanation. Modified.